GVFS/GVFS.Common/ModifiedPathsDatabase.cs

```diff
@@ -183,6 +185,22 @@ private bool TryParseRemoveLine(string line, out string key, out string error)
             return true;
         }
 
+        private bool ContainsParentDirectory(string modifiedPath)
+        {
+            string[] pathParts = modifiedPath.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+            StringBuilder parentFolder = new StringBuilder();

```

Comment:
I think this `StringBuilder` is actually causing us to do extra work here, since we have to convert every path to a string anyway on line 195. I would just use normal string concatenation in this code.